x/stakeibc/keeper/send_host_zone_unbondings_test.go

```diff
@@ -0,0 +1,164 @@
+	stakeibc "github.com/Stride-Labs/stride/x/stakeibc/types"
+)
+
+type SendHostZoneUnbondingTestCase struct {
+	amt_to_unbond           uint64
+	epoch_unbonding_records []recordtypes.EpochUnbondingRecord
+	host_zone               stakeibc.HostZone
+	lightClientTime         uint64
+}
+
+func (s *KeeperTestSuite) SetupSendHostZoneUnbonding() SendHostZoneUnbondingTestCase {
+	host_val1_addr := "cosmos_VALIDATOR_1"
+	host_val2_addr := "cosmos_VALIDATOR_2"
+	delegationAddr := "cosmos_DELEGATION"
+	amt_to_unbond := uint64(1_000_000)
+	amt_val1 := uint64(2_000_000)
+	amt_val2 := uint64(1_000_000)
+	wgt_val1 := uint64(2)
+	wgt_val2 := uint64(1)
+	unbonding_time := uint64(1660348276)
+	light_client_time := unbonding_time + 1
+
+	//  define the host zone with stakedBal and validators with staked amounts
+	validators := []*stakeibc.Validator{
+		{
+			Address:       host_val1_addr,
+			DelegationAmt: amt_val1,
+			Weight:        wgt_val1,
+		},
+		{
+			Address:       host_val2_addr,
+			DelegationAmt: amt_val2,
+			Weight:        wgt_val2,
+		},
+	}
+
+	delegationAccount := stakeibc.ICAAccount{
+		Address: delegationAddr,
+		Target:  stakeibc.ICAAccountType_DELEGATION,
+	}
+
+	hostZone := stakeibc.HostZone{
+		ChainId:           "GAIA",
+		HostDenom:         "uatom",
+		Bech32Prefix:      "cosmos",
+		Validators:        validators,
+		DelegationAccount: &delegationAccount,
+	}
+
+	// list of epoch unbonding records
+	epochUnbondingRecords := []recordtypes.EpochUnbondingRecord{
+		recordtypes.EpochUnbondingRecord{
+			Id:                   1,
+			UnbondingEpochNumber: 1,
+			HostZoneUnbondings:   make(map[string]*recordtypes.HostZoneUnbonding),
+		},
+	}
+
+	// for each epoch unbonding record, add a host zone unbonding record and append the record
+	for _, epochUnbondingRecord := range epochUnbondingRecords {
+		// note: we're using the same hostZoneUnbonding for each epoch unbonding record in this test
+		epochUnbondingRecords[0].HostZoneUnbondings["GAIA"] = &recordtypes.HostZoneUnbonding{
+			Amount:        amt_to_unbond,
+			Denom:         "uatom",
+			HostZoneId:    "GAIA",
+			UnbondingTime: unbonding_time, // 2022-08-12T19:51
+			Status:        recordtypes.HostZoneUnbonding_BONDED,
+		}
+
+		s.App.RecordsKeeper.AppendEpochUnbondingRecord(s.Ctx, epochUnbondingRecord)
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	return SendHostZoneUnbondingTestCase{
+		amt_to_unbond:           amt_to_unbond,
+		host_zone:               hostZone,
+		epoch_unbonding_records: epochUnbondingRecords,
+		lightClientTime:         light_client_time, // 2022-08-13T19:51, 1d after the unbonding time
+	}
+}
+
+func (s *KeeperTestSuite) TestSendHostZoneUnbondingSuccessful() {
+	tc := s.SetupSendHostZoneUnbonding()
+
+	msgs, totalAmtToUnbond, err := s.App.StakeibcKeeper.SendHostZoneUnbondings(s.Ctx, tc.host_zone)
+	s.Require().Nil(err)
+
+	// function to get total wgt of all validators on a host zone
+	getTotalWgt := func(hostZone stakeibc.HostZone) sdk.Dec {
```

Comment:
might be easier to just return this value from SetupSendHostZoneUnbondings instead of calculating it (since there's only two validators)